### PR TITLE
feat(big-query): Realign transaction v1 legacy schema

### PR DESCRIPTION
**Pull request by eytand**

### CODE REVIEW

File: lib/v1/transactions/legacy/base.js [hunk 1]
```diff
@@ -276,20 +240,11 @@ module.exports = {
           type: 'string',
           format: 'uuid',
           example: '09a9a469-c16b-4b54-835d-692d05d218f9',
-          description: 'If sfes are involved, the Tillhub resource identifier of that safe'
+          description: 'If safes are involved, the Tillhub resource identifier of that safe (currently ignored by the API)'
         }
       }
     })
   },
-  temp_staff: {
```
Comment (alghanor) [inline lib/v1/transactions/legacy/base.js:284]: this is used and should stay... hard to tell if this ever happens (but in theory it's possible)

File: lib/v1/transactions/legacy/base.js [hunk 2]
```diff
@@ -313,7 +268,18 @@ module.exports = {
...
-        type: 'object'
+        type: 'object',
+        properties: {
+          name: {
+            type: 'string'
+          },
+          value: {
+            type: 'string'
+          },
+          fr_transaction_id: {
```
Comment (alghanor) [inline lib/v1/transactions/legacy/base.js:279]: I thought this is just [name: value], fr_transaction_id is not used from our side

File: lib/v1/transactions/legacy/base.js [hunk 3]
```diff
@@ -381,14 +342,6 @@ module.exports = {
     The Tillhub staff resource ID, that has generated this transaction.
     `
   }),
-  _temporary_staff: oneOf({
```
Comment (alghanor) [inline lib/v1/transactions/legacy/base.js:384]: should stay... is meant to replace temporary_staff

File: lib/v1/transactions/legacy/base.js [hunk 4]
```diff
@@ -323,11 +289,6 @@ module.exports = {
     default: false,
     description: 'If true, the transaction contains items modified by the promotion engine'
   }),
-  branch: oneOf({
```
Comment (alghanor) [inline lib/v1/transactions/legacy/base.js:326]: should stay... not sure if a register can ever switch to another branch... but who knows?

File: lib/v1/transactions/legacy/items/base.js [hunk 5]
```diff
@@ -348,24 +355,6 @@ module.exports = {
             type: 'boolean',
             default: false,
             description: 'If true, this discount was trigger automatically.'
-          }),
-          order: oneOf({
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:352]: this must stay.. it belongs to discounts -> txCiDiscounts

File: lib/v1/transactions/legacy/items/base.js [hunk 6]
```diff
@@ -348,24 +355,6 @@ module.exports = {
...
-            type: 'string',
-            enum: [
-              'first',
-              'last'
-            ],
-            default: 'last'
-          }),
-          vat_rate_class: oneOf({
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:360]: yes, this does not belong to discounts

**Review by alghanor — CHANGES_REQUESTED** [review]
see comments

File: lib/v1/transactions/legacy/base.js [hunk 7]
```diff
@@ -93,33 +87,19 @@ module.exports = {
...
+      type: 'number',
       example: 3500,
       description: 'The total amount of this transaction in the lowest (available) fraction of that currency -> 35.00 EUR or 3500 HUF respectively'
     })
   },
   change: {
     deprecated: true,
     ...oneOf({
-      type: 'integer',
+      type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/base.js:98]: must be INT

File: lib/v1/transactions/legacy/items/base.js [hunk 8]
```diff
@@ -15,42 +15,45 @@ const customProperties = require('../../../../common/custom_properties')
...
+    type: 'string',
     ...oneOf({
       description: 'Apparently not used'
     })
   },
   article_price: {
     deprecated: true,
     ...oneOf({
-      type: 'integer',
+      type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:40]: must be INT

File: lib/v1/transactions/legacy/items/base.js [hunk 9]
```diff
@@ -15,42 +15,45 @@ const customProperties = require('../../../../common/custom_properties')
...
       example: 799,
       description: 'The per qty=1 raw price of the product in the lowest (available) fraction of that currency -> 7.99 EUR or 799 HUF respectively'
     })
   },
   selling_price: {
     deprecated: true,
     ...oneOf({
-      type: 'integer',
+      type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:48]: must be INT

File: lib/v1/transactions/legacy/items/base.js [hunk 10]
```diff
@@ -15,42 +15,45 @@ const customProperties = require('../../../../common/custom_properties')
...
       example: 612,
       description: 'The per qty=1 (disocunted) price of the product in the lowest (available) fraction of that currency -> 6.12 EUR or 612 HUF respectively'
     })
   },
   selling_price_total: {
     deprecated: true,
     ...oneOf({
-      type: 'integer',
+      type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:56]: must be INT

File: lib/v1/transactions/legacy/items/base.js [hunk 11]
```diff
@@ -82,7 +85,7 @@ module.exports = {
   vat_amount: {
     deprecated: true,
     ...oneOf({
-      type: 'integer',
+      type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:88]: must be INT

File: lib/v1/transactions/legacy/items/base.js [hunk 12]
```diff
@@ -118,7 +123,7 @@ module.exports = {
   discount_amount: {
     deprecated: true,
     ...oneOf({
-      type: 'integer',
+      type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:126]: must be INT

File: lib/v1/transactions/legacy/items/base.js [hunk 13]
```diff
@@ -160,7 +167,7 @@ module.exports = {
   is_refund: {
     deprecated: true,
     ...oneOf({
-      type: 'number',
+      type: 'integer',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:170]: should be BOOL, please verify

File: lib/v1/transactions/legacy/items/base.js [hunk 14]
```diff
@@ -317,12 +324,12 @@ module.exports = {
             description: 'Source discount info, depending on type this means for \'percentage\': 25.137 % and for \'value\': 0.25137 EUR (sic!) or 25.137 HUF (sic!) respectively'
           },
           amount: {
-            type: 'integer',
+            type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:327]: must be INT

File: lib/v1/transactions/legacy/items/base.js [hunk 15]
```diff
@@ -317,12 +324,12 @@ module.exports = {
...
             example: 43,
             description: 'The per qty=1 effective discount amount -> 0.43 EUR or 43 HUF respectively'
           },
           amount_total: {
-            type: 'integer',
+            type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:332]: must be INT

File: lib/v1/transactions/legacy/payments/base.js [hunk 16]
```diff
@@ -19,7 +19,7 @@ module.exports = {
   amount: {
     deprecated: true,
     ...oneOf({
-      type: 'integer',
+      type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/payments/base.js:22]: must be INT

File: lib/v1/transactions/legacy/payments/base.js [hunk 17]
```diff
@@ -101,7 +101,7 @@ module.exports = {
   tip: {
     deprecated: true,
     ...oneOf({
-      type: 'integer',
+      type: 'number',
```
Comment (alghanor) [inline lib/v1/transactions/legacy/payments/base.js:104]: must be INT

**Review by alghanor — CHANGES_REQUESTED** [review]
see comments

File: lib/v1/transactions/legacy/base.js (hunk 7 above)
Comment (eytand) [reply lib/v1/transactions/legacy/base.js]: fixed

File: lib/v1/transactions/legacy/base.js (hunk 2 above)
Comment (eytand) [reply lib/v1/transactions/legacy/base.js]: removed

File: lib/v1/transactions/legacy/items/base.js (hunk 8 above)
Comment (eytand) [reply lib/v1/transactions/legacy/items/base.js]: fixed

File: lib/v1/transactions/legacy/items/base.js (hunk 9 above)
Comment (eytand) [reply lib/v1/transactions/legacy/items/base.js]: fixed

File: lib/v1/transactions/legacy/items/base.js (hunk 10 above)
Comment (eytand) [reply lib/v1/transactions/legacy/items/base.js]: fixed

File: lib/v1/transactions/legacy/items/base.js (hunk 11 above)
Comment (eytand) [reply lib/v1/transactions/legacy/items/base.js]: fixed

File: lib/v1/transactions/legacy/items/base.js (hunk 12 above)
Comment (eytand) [reply lib/v1/transactions/legacy/items/base.js]: fixed

File: lib/v1/transactions/legacy/items/base.js (hunk 14 above)
Comment (eytand) [reply lib/v1/transactions/legacy/items/base.js]: fixed

File: lib/v1/transactions/legacy/items/base.js (hunk 15 above)
Comment (eytand) [reply lib/v1/transactions/legacy/items/base.js]: fixed

File: lib/v1/transactions/legacy/payments/base.js (hunk 16 above)
Comment (eytand) [reply lib/v1/transactions/legacy/payments/base.js]: fixed

File: lib/v1/transactions/legacy/payments/base.js (hunk 17 above)
Comment (eytand) [reply lib/v1/transactions/legacy/payments/base.js]: fixed

File: lib/v1/transactions/legacy/base.js (hunk 3 above)
Comment (eytand) [reply lib/v1/transactions/legacy/base.js]: re-adding. The removed items are ignored by the API and never get persisted

File: lib/v1/transactions/legacy/items/base.js (hunk 5 above)
Comment (eytand) [reply lib/v1/transactions/legacy/items/base.js]: not being persisted

**Review by eytand — COMMENTED** [review]
@alghanor fixed according to your comments

File: lib/v1/transactions/legacy/items/base.js [hunk 18]
```diff
@@ -379,6 +376,9 @@ module.exports = {
       items: {
         type: 'object',
         properties: {
+          fr_transaction_cartitem_id: oneOf({
```
Comment (alghanor) [inline lib/v1/transactions/legacy/items/base.js:379]: this should not be here

**Review by alghanor — CHANGES_REQUESTED** [review]
fr_transaction_cartitem_id should not be in items.base.context

**Review by alghanor — APPROVED** [review]
approved

**Pull request merged**